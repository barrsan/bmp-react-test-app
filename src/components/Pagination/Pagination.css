.nav-item_active {
  background-color: gainsboro;
  border-radius: 99%;
}