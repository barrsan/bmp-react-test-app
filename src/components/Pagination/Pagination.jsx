import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import './Pagination.css';


const propTypes = {
  pagesNumber: PropTypes.number,
  currentPage: PropTypes.number,
};

const defaultProps = {
  pagesNumber: 0,
  currentPage: 0,
};


const Pagination = ({
  pagesNumber,
  currentPage,
}) => {
  const pages = [];

  if (pagesNumber) {
    for (let i = 1; i < pagesNumber + 1; i += 1) {
      const item = (
        <li
          className={classNames({
            'nav-item': true,
            'nav-item_active': i === currentPage,
          })}
          key={i}
        >
          <Link
            className="nav-link"
            to={{
              search: `?page=${i}`,
            }}
          >
            {i}
          </Link>
        </li>
      );
      pages.push(item);
    }
  }

  return (
    <ul className="pagination nav justify-content-center">
      {pages}
    </ul>
  );
};


Pagination.propTypes = { ...propTypes };
Pagination.defaultProps = { ...defaultProps };


export default Pagination;
