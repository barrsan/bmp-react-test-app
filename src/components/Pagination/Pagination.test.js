import React from 'react';
import { shallow } from 'enzyme';
import Pagination from './Pagination';


describe('Pagination component', () => {
  describe('Have 2 pages', () => {
    const props = {
      pagesNumber: 2,
      currentPage: 2,
    };

    const wrapper = shallow(<Pagination {...props} />);

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());

    test('renders without crashing', () => {
      expect(wrapper).toHaveLength(1);
    });

    test('renders 2 links', () => {
      expect(wrapper.find('Link')).toHaveLength(2);
    });

    test('renders link with .nav-item_active class', () => {
      expect(wrapper.find('.nav-item_active')).toHaveLength(1);
    });
  });


  describe('Have 0 pages', () => {
    const props = {
      pagesNumber: 0,
    };

    const wrapper = shallow(<Pagination {...props} />);

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());

    test('renders without crashing', () => {
      expect(wrapper.find('.pagination')).toHaveLength(1);
    });

    test('renders 0 links', () => {
      expect(wrapper.find('Link')).toHaveLength(0);
    });
  });
});
