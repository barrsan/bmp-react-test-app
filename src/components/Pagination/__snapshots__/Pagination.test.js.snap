// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Have 0 pages renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    currentPage={0}
    pagesNumber={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "pagination nav justify-content-center",
    },
    "ref": null,
    "rendered": Array [],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "pagination nav justify-content-center",
      },
      "ref": null,
      "rendered": Array [],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Pagination component Have 2 pages renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    currentPage={2}
    pagesNumber={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="nav-item"
>
          <Link
                    className="nav-link"
                    replace={false}
                    to={
                              Object {
                                        "search": "?page=1",
                                      }
                    }
          >
                    1
          </Link>
</li>,
        <li
          className="nav-item nav-item_active"
>
          <Link
                    className="nav-link"
                    replace={false}
                    to={
                              Object {
                                        "search": "?page=2",
                                      }
                    }
          >
                    2
          </Link>
</li>,
      ],
      "className": "pagination nav justify-content-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="nav-link"
            replace={false}
            to={
                        Object {
                                    "search": "?page=1",
                                  }
            }
>
            1
</Link>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 1,
            "className": "nav-link",
            "replace": false,
            "to": Object {
              "search": "?page=1",
            },
          },
          "ref": null,
          "rendered": 1,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="nav-link"
            replace={false}
            to={
                        Object {
                                    "search": "?page=2",
                                  }
            }
>
            2
</Link>,
          "className": "nav-item nav-item_active",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 2,
            "className": "nav-link",
            "replace": false,
            "to": Object {
              "search": "?page=2",
            },
          },
          "ref": null,
          "rendered": 2,
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-item"
>
            <Link
                        className="nav-link"
                        replace={false}
                        to={
                                    Object {
                                                "search": "?page=1",
                                              }
                        }
            >
                        1
            </Link>
</li>,
          <li
            className="nav-item nav-item_active"
>
            <Link
                        className="nav-link"
                        replace={false}
                        to={
                                    Object {
                                                "search": "?page=2",
                                              }
                        }
            >
                        2
            </Link>
</li>,
        ],
        "className": "pagination nav justify-content-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="nav-link"
              replace={false}
              to={
                            Object {
                                          "search": "?page=1",
                                        }
              }
>
              1
</Link>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 1,
              "className": "nav-link",
              "replace": false,
              "to": Object {
                "search": "?page=1",
              },
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="nav-link"
              replace={false}
              to={
                            Object {
                                          "search": "?page=2",
                                        }
              }
>
              2
</Link>,
            "className": "nav-item nav-item_active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 2,
              "className": "nav-link",
              "replace": false,
              "to": Object {
                "search": "?page=2",
              },
            },
            "ref": null,
            "rendered": 2,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
